File: src/app/Models/product.ts
import { Review } from "./review";

export class Product {
    _id: string;
    productName: string;
    aboutProduct: string; 
    sellerId: string; 
    brand: string;
    maxRetailPrice: number;
    sellingPrice: number;
    quantity: number;
    unitsSold: number
    productCategories: string[]; 
    averageRating:  number;
    productImages: string[];   
    reviews: Review[];

    constructor(
        _id: string,
        productName: string,
        aboutProduct: string,
        sellerId: string, 
        brand: string,
        productCategories: string[],
        maxRetailPrice: number,
        sellingPrice: number,
        quantity: number,
        unitsSold: number, 
        averageRating:  number,
        productImages: string[], 
        reviews: Review[]
    ) {
        this._id = _id;
        this.productName = productName;
        this.aboutProduct = aboutProduct;
        this.sellerId = sellerId;
        this.brand = brand;
        this.maxRetailPrice = maxRetailPrice;
        this.sellingPrice = sellingPrice;
        this.quantity = quantity;
        this.unitsSold = unitsSold;
        this.productCategories = productCategories;
        this.averageRating = averageRating;
        this.productImages = productImages;
        this.reviews = reviews;
    }

}

File: GoServer/Model/product.go
package Model

 
type Product struct {
  Id        			    string      `json:"_id" bson:"_id"`
	ProductName			    string 		  `json:"productName" bson:"productName"`
	AboutProduct 		    string 		  `json:"aboutProduct" bson:"aboutProduct"`
	SellerId			      string		  `json:"sellerId" bson:"sellerId"`
	Brand				        string		  `json:"brand" bson:"brand"`
	MaxRetailPrice		  int 		    `json:"maxRetailPrice" bson:"maxRetailPrice"`
	SellingPrice		    int			    `json:"sellingPrice" bson:"sellingPrice"`
	Quantity			      int 		    `json:"quantity" bson:"quantity"`
	UnitsSold			      int 		    `json:"unitsSold" bson:"unitsSold"`
	ProductCategories	  []string 	  `json:"productCategories" bson:"productCategories"`
	AverageRating		    int 		    `json:"averageRating" bson:"averageRating"`
	ProductImages 		  []string 	  `json:"productImages" bson:"productImages"`
	ProductProperties   []string 	  `json:"productProperties" bson:"productProperties"`
	Reviews				      []Review	  `json:"reviews" bson:"reviews"`
}
